pkg/kubernetes/scripts/kube-client.js

```diff
@@ -64,6 +64,7 @@
     ]);
 
     var NAME_RE = /^[a-z0-9]([-a-z0-9]*[a-z0-9])?$/;
```

Comment:
Yes, there currently is neither a request nor a good reason to allow email-style names for e. g. cluster or node names. I think it's safer to keep this narrow for user names only.